File: src/TadbirNG/SPPC.Tadbir.Interfaces/Api/UserValueApi.cs
using System;

namespace SPPC.Tadbir.Api
{
    /// <summary>
    /// Defines API server routes and API client operation URLs for working with user values
    /// </summary>
    public sealed class UserValueApi
    {
        /// <summary>
        /// API client URL for all categories of user values
        /// </summary>
        public const string Categories = "user-values/categories";

        /// <summary>
        /// API server route URL for all categories of user values
        /// </summary>
        public const string CategoriesUrl = "user-values/categories";

        /// <summary>
        /// API client URL for all user values in a category specified by unique identifier
        /// </summary>
        public const string CategoryValues = "user-values/categories/{0}/values";

        /// <summary>
        /// API server route URL for all user values in a category specified by unique identifier
        /// </summary>
        public const string CategoryValuesUrl = "user-values/categories/{categoryId:min(1)}/values";

        /// <summary>
        /// API client URL for all user values
        /// </summary>
        public const string UserValues = "user-values/values";

        /// <summary>
        /// API server route URL for all user values
        /// </summary>
        public const string UserValuesUrl = "user-values/values";

        /// <summary>
        /// API client URL for a single user value specified by unique identifier
        /// </summary>
        public const string UserValue = "user-values/values/{0}";

        /// <summary>
        /// API server route URL for a single user value specified by unique identifier
        /// </summary>
        public const string UserValueUrl = "user-values/values/{valueId:min(1)}";
    }
}

File: src/TadbirNG/SPPC.Tadbir.Web/ClientApp/src/app/shared/services/api/userValueApi.ts
import { environment } from "@sppc/env/environment";

export class UserValueApi {

    // user-values/categories
    public static Categories = environment.BaseUrl + "/user-values/categories";

    // user-values/categories/{categoryId:min(1)}/values
    public static CategoryValues = environment.BaseUrl + "/user-values/categories/{0}/values";

    // user-values/values
    public static UserValues = environment.BaseUrl + "/user-values/values";

    // user-values/values/{valueId:min(1)}
    public static UserValue = environment.BaseUrl + "/user-values/values/{0}";
}
